src/lib/dataEntities/networks.ts

```diff
@@ -329,9 +349,87 @@ export const l2Networks: L2Networks = {
   },
 }
 
-const getNetwork = async (
+/** Determines if a chain is a parent of *any* other chain. */
+const isParentChain = (chain: Chain): chain is ParentChain => {
+  return (
+    chain &&
+    'partnerChainIDs' in chain &&
+    Array.isArray(chain.partnerChainIDs) &&
+    chain.partnerChainIDs.length > 0
+  )
+}
+
+/** Determines if a chain is a child of *any* other chain. */
+const isChildChain = (chain: Chain): chain is ChildChain => {
+  return chain && 'partnerChainID' in chain
+}
+
+const isL1Chain = (chain: Chain): chain is L1Network => {
+  return chain && isParentChain(chain) && !isChildChain(chain)
+}
+
+const isL2Chain = (chain: Chain): chain is L2Network => {
+  return chain && isChildChain(chain) && !isOrbitChain(chain)
+}
+
+const isOrbitChain = (chain: Chain): chain is OrbitChain => {
+  return chain && 'isOrbit' in chain && chain.isOrbit
+}
+
+/**
+ * Builds an object that is a list of chains filtered by the provided predicate function indexed by their chain id
+ * @param filterFn - A predicate function to determine if a chain should be included.
+ * @return An object with only the filtered chains.
+ */
+const getChainsByType = <T extends typeof networks>(
+  filterFn: (chain: Chain) => boolean
+): T => {
+  return Object.entries(networks).reduce<typeof networks>(
+    (accumulator, [chainId, chainData]) => {
+      if (filterFn(chainData)) {
+        accumulator[chainId] = chainData
+      }
+      return accumulator
+    },
+    {}
+  ) as T
+}
+
+const getL1Chains = () => getChainsByType<L1Networks>(isL1Chain)
+const getL2Chains = () => getChainsByType<L2Networks>(isL2Chain)
+const getChildChains = () => getChainsByType<ChildChains>(isChildChain)
+const getParentChains = () => getChainsByType<ParentChains>(isParentChain)
+
+export const getParentForNetwork = (chain: Chain) => {
+  if (!isChildChain(chain)) {
+    return undefined
+  }
+  const parentChain = networks[chain.partnerChainID]
+  if (!parentChain || !isParentChain(parentChain)) {
+    throw new ArbSdkError(
+      `ParentChain ${chain.partnerChainID} not recognized for Chain ${chain.chainID}.`
+    )
+  }
+  return parentChain
+}
+
+const getChildrenForNetwork = (chain: Chain) => {
+  if (!isParentChain(chain)) {
+    return undefined
+  }
+  const childChains = getChildChains()
+  const children = Object.values(childChains).filter(
+    childChain => childChain.partnerChainID === chain.chainID
+  )
+  return children
+}
+
+export let l1Networks: L1Networks = getL1Chains()
+export let l2Networks: L2Networks = getL2Chains()
+
+export const getNetwork = async (
   signerOrProviderOrChainID: SignerOrProvider | number,
-  layer: 1 | 2
+  layer?: 1 | 2
```

Comment:
We should keep this required but treat it as "parent chain" and "child chain".

If you pass in 1, then `getL1Network` should work for both an L1 chain and an L2 that is a parent chain for Orbit chains.
If you pass in 2, then `getL2Network` should work for either an L2 or an Orbit chain.

That makes it work with all existing code without any changes. I know it's confusing in the interim before everything is renamed.